CONTRIBUTING.md

```diff
@@ -0,0 +1,112 @@
+Contribute to Che-Theia
+================
+
+There are 3 contribution points:
+- Theia
+- Che-Theia extensions
+- Che-Theia plugins
+
+For each of these kinds of contribution, rebuilding the container image is not needed to code, build and test your contribution.
+We can create a Che workspace that would setup all the containers to build and run Che-Theia. The containers will rely on the same ones used in the CI for the Che-Theia container build. Building che-theia in the che-workspace would be very close to the one build by Docker.
+
+You will find here few devfile that will help you in setting up theses environments
+
+## Devfiles
+Each devfile could be run on any Che instances. Beware that each of them may need a certain amount of memory.
+Devfile could be launched through a factory or [chectl](https://github.com/che-incubator/chectl) cli.
+
+```
+$ chectl workspace:start -f devfiles/che-theia-all.devfile.yaml
+```
+
+## Devfile for contributing to Theia or Che-Theia extensions or Che-plugins
+> Note that this devfile is going to evolve very soon and be splitted into multiple devfiles.
+
+The devfile: [<this.repo>/devfiles/che-theia-all.devfile.yaml](../devfiles/che-theia-all.devfile.yaml)
+
+The goal is to show how to setup a che7 environment to work on che-theia.
```

Comment:
In this section, we show how to setup a Che 7 environment to work on che-theia, and how to use it.